src/averageAges.js

```diff
@@ -55,7 +73,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const listOfChildren = people.filter(child =>
+    people.find(person => person.name === child.mother));
+
+  const pickChildren = onlyWithSon
+    ? listPeopleBySex(listOfChildren, 'm')
+    : listOfChildren;
```

Comment:
here you can use one variable and check for `onlyWithSon` using the conditional operator `?`